### ISSUE TITLE
[BUG] Caching prevents files to be loaded even after file change

### ISSUE THREAD

**Issue by bpannier** [mnaoumov/obsidian-fix-require-modules#4]
### Description

This plugin enables me to use Typescripts as the base for everything I do with Dataview, thank you for this!

Also it seems that files which are included via "require" will not be reloaded even when they have changed. Sometimes a new version will be used once and on the next execute an old version will be used.

Would be good to have a command to invalidate the cache, turning off the cache completely, or just have a working cache ;)

### Steps to Reproduce

In a Note:

\`\`\`dataviewjs 
const { render } = require("/dist/myscript.js?force");
console.log("HOO1");
render(dv);
console.log("HOO2"); 
\`\`\`

ts/myscript.ts (which will create dist/myscript.js via tsc as soon this changes):

```typescript
import { MyClass } from "./MyClass";
export function render(dv : any) {
    console.log("DD in render()");
    MyClass.render(dv);
}
```

ts/MyClass.ts (which will create dist/MyClass.js via tsc as soon this changes):

```typescript
export class MyClass {
   static render(dv:any) {
        console.log("EE in MyClass::render()");
   }
}
```

When MyClass.ts / MyClass.js changes, it will not be loaded again, and therefore the cached version will be used for ever.

I tried to disable and enable the plugin, but that did not invalidate the cache either. Only restarting Obsidian helps.

### Expected Behavior

Expected is that when a file changes, independent of where it is or how deep the inclusion is in the tree, the cache will be invalidated and the file will be reloaded. 

### Actual Behavior

The file will not be reloaded even when it has changed.

### Environment Information

- **Plugin Version**: 5.0.5
- **Obsidian Version**: 1.7.4
- **Operating System**: MacOS 15.0.1
- **Dataview**: 0.5.67

### Attachments

Typescript configuration: [tsconfig.json](https://github.com/user-attachments/files/17599566/tsconfig.json)

### Confirmations

- [X] I attached a video showing the issue
- [X] I attached a sample vault where the issue can be reproduced
- [X] I have tested the issue with the latest version
- [X] I have checked for existing issues

**Comment by mnaoumov** [comment]
@bpannier I don't see video and vault attached

**Comment by bpannier** [comment]
Vault: [Fix Require.zip](https://github.com/user-attachments/files/17601286/Fix.Require.zip)

Sorry, but I will not create a video. Feel free to edit the console output in MyClass after the first successful run, and compile it. You will see that the output in the console will not change.

**Comment by mnaoumov** [comment]
Fixed in https://github.com/mnaoumov/obsidian-fix-require-modules/releases/tag/5.1.0

Also I would like to ask about your workflow. Why do you need to compile `ts` to `js`? With this plugin your can just 

```ts
require('ts/myscript.ts')
```

**Comment by bpannier** [comment]
You rock, thank you!

When try to include the TS files directly I get:

```
Evaluation Error: TransformError: spawnSync /Users/bpannier/Library/CloudStorage/GoogleDrive-bpannier@gmail.com/My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/@esbuild/darwin-arm64/bin/esbuild EACCES
    at Object.spawnSync (node:internal/child_process:1124:20)
    at spawnSync (node:child_process:891:24)
    at Object.execFileSync (node:child_process:934:15)
    at Object.func [as execFileSync] (node:electron/js2c/node_init:2:2214)
    at runServiceSync (/Users/bpannier/Library/CloudStorage/GoogleDrive-bpannier@gmail.com/My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/esbuild/lib/main.js:2084:30)
    at transformSync (/Users/bpannier/Library/CloudStorage/GoogleDrive-bpannier@gmail.com/My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/esbuild/lib/main.js:1908:3)
    at eval (plugin:fix-require-modules:76522:45)
    at applyTransformersSync (plugin:fix-require-modules:76487:16)
    at transformSync (plugin:fix-require-modules:76518:22)
    at Object.transformer [as .js] (plugin:fix-require-modules:76659:18)
```
